### ISSUE TITLE
Go through all "original" issues

### ISSUE THREAD

**Issue by toxicphreAK** [toxicphreAK/python-docx-ng#25]
Go through all issues: https://github.com/python-openxml/python-docx/issues

**Comment by toxicphreAK** [comment]
Swept the upstream tracker (371 open issues) and pull requests (132 open) and filed the outcome as #64–#87.

**Method.** Ranked upstream issues by engagement, then checked each candidate against the actual 2.0.0 base rather than trusting the upstream description — several long-standing upstream complaints are already fixed here and were dropped:

- `pathlib.Path` for open *and* save already works (upstream PR 1571) — only the type annotation on `Document()` omits it
- fractional `w:pgMar` twips already parse (upstream PR 1552)
- images with a zero-DPI header already fall back cleanly (upstream 515, PR 1505)
- `cell._tc.bottom` on `vMerge` + `gridSpan` does not raise here (upstream 1458, PR 1526)
- ragged rows, style lookup by UI name, and duplicate-zip-member warnings on save are all fine
- `_Cell.grid_span` and `ParagraphFormat.outline_level` are already present

Skipped as already tracked: footnotes (#34), SVG/EMF/WMF (#35), form fields (#36), AltChunk (#37), table borders (#38).

**Bugs — all reproduced on this base:**

| | |
| --- | --- |
| #64 | dangling relationship makes a document unopenable (`KeyError`) |
| #65 | corrupt/truncated/encrypted package raises `BadZipFile` |
| #66 | `w:highlight w:val="none"` raises |
| #67 | non-integer `w:sz` raises |
| #68 | missing `w:tblGrid` makes a table unreadable |
| #69 | `Table.cell()` is quadratic — 199 s for a 200×10 table |
| #70 | text inside `w:sdt` content controls is silently invisible |

#69 and #70 were not on the shortlist going in; both were found by measuring rather than reading. #70 is silent data loss on read.

**Features upstream has never shipped:** #71 hyperlink writing, #72 custom properties, #73 list numbering, #74 tracked changes, #75 bookmarks, #76 search/replace, #77 deletion API, #78 section columns, #79 CJK/complex-script fonts, #80 fields, #81 alt text, #82 WebP, #83 `.dotx`, #84 vertical merge extent, #85 floating images, #86 watermark, #87 style copying.

Dependency order worth noting: #75 (bookmarks) gates the referential half of #80 (fields), which is what #71's internal-anchor form, captions and a refreshable TOC all need. #70 should land before #76, or replace will silently skip template placeholders.

Left unmilestoned — none of these are rebase work, so they need triage against the 2.0.0 scope rather than inheriting it.

**Comment by toxicphreAK** [comment]
Reviewed in the 2.0.0 feature pass (August 2026) and left open deliberately: this is a standing research task rather than a unit of implementable work, so there is nothing here to close.

Several of the features these threads point at have since landed on `main` independently — tracked changes (#74), fields and TOC (#80), list numbering (#73), cross-run search and replace (#76), floating images (#85), watermarks (#86) and style copying (#87). Worth re-reading this list against HISTORY.rst before spending time on it, since the overlap is now substantial.

**Comment by toxicphreAK** [comment]
Went back over the upstream tracker to check whether this could be closed. The earlier sweep on this thread was **ranked by engagement**, which is the right way to spend a first pass but leaves a tail unread — and the tail turned out to have real material in it. Eight more issues filed; now it can close.

371 open upstream issues, unchanged since the last sweep (nothing new opened since 2026-08-01). This pass read the whole distribution: 58 issues at 10+ comments and reactions, 53 at 5–9, 112 at 2–4, and 148 at 0–1.

### Filed from the tail

| | | Upstream |
|---|---|---|
| #119 | Text in text boxes and `w:smartTag` silently dropped from `Paragraph.text` | "Drawings are not parsed…", "text in the tag 'smartTag'" |
| #120 | ISO Strict OOXML documents cannot be opened | "Unable to read the strict open xml Docx file" |
| #121 | Style names containing a quote break XPath lookup | "escape style names to allow embedded quotes" |
| #122 | `Section.orientation` does not swap page dimensions | filed upstream at least 3 times |
| #123 | Paragraph borders (`w:pBdr`) and page borders | "feature: horz rule (paragraph border)" |
| #124 | `InlineShape.image` | "feature: InlineShape.image ?" |
| #125 | Honour EXIF orientation when inserting images | "Problem with rotation when adding images", PR #1575 |
| #126 | Charts — a written "not planned", rather than a third silent drop | 31 and 16 engagement, two separate issues |

All four bugs reproduce on `main`. The two worth singling out:

- **#119** is silent data loss of the same class as #70 and #95. A pull quote or callout box contributes nothing to `Paragraph.text`, and `w:smartTag` hides its runs from `.runs` as well. It also means `replace_text()` never substitutes a placeholder inside a text box — a template comes back unchanged with no error.
- **#120** fails with `AttributeError: 'lxml.etree._Element' object has no attribute 'body'`, which tells the user nothing. Even just detecting the Strict namespace and raising something that names it would be worth having.

### Checked and *not* filed, because they do not reproduce here

Each of these is a live upstream complaint that the 2.0.0 base already handles:

- **"Setting `core_properties.last_modified_by` makes document invalid"** (19 engagement) — the accessors use `successors=()` and append at the end, which looks wrong. It is not: `ref/xsd/opc-xsd/opc-coreProperties.xsd:16` declares `CT_CoreProperties` as `xs:all`, so child order is unconstrained. Same trap as `CT_Properties` in app.xml, which is why the schema is worth checking before assuming child order matters.
- **"Large Table Creation Slow"** (21) — `add_table(200, 10)` is instant, 2000 cell writes take 0.18 s, `cell(199, 9)` takes 0.2 ms. Fixed by #69.
- **`Row.height` / `allow_break_across_pages`** — `_Row.height`, `.height_rule` and `.dont_split` all present.
- **"Cannot access Heading 1 style"** (17) — that specific failure is #110, already filed from the PR sweep.
- **Zero-DPI `add_picture` `ZeroDivisionError`**, `cell._tc.bottom` `ValueError`, NULL relationships, missing `w:tblGrid`, ragged rows, `.docm`, WebP, checkbox form fields, `Font.cs_size`, `Font.name_far_east` — all already handled.

### Mapping the rest of the high-engagement tail to what exists

Hyperlinks → #71 · restart numbering, nested lists, `get_listnum` → #73 · custom properties → #72 · search/replace and `isolate_run` → #76 · bookmarks → #75 · cross-references, fields, TOC/LOF/LOT, page numbers → #80 · captions → #102 · MathML/equations → #95 · track changes → #74 · `Paragraph.delete` / `Row.delete` → #77 · `w:sdt` → #70 · floating images → #85 · watermark → #86 · columnar layout → #78 · CJK typefaces → #79 · copy styles → #87 and #116 · copy table across documents, copy a run's font, "add row with formatting" → #112 · `Font.shading` → #103 · repeat header row → #106 · RTL in headers → #108 · default font for a document → #99 · table borders → #38 · **"Inconsistent document icons"** → #113, which removes the template thumbnail that causes it.

Genuinely out of scope and left alone: Markdown conversion, EPS/EMS image formats, detecting page breaks (a rendering question), and SpreadsheetML.

Nothing left to track here. Closing — the work is now in #94–#126.
